dev/devicelab/lib/framework/cocoon.dart

```diff
@@ -77,35 +77,28 @@ class Cocoon {
   /// 1. Run DeviceLab test, writing results to a known path
   /// 2. Request service account token from luci auth (valid for at least 3 minutes)
   /// 3. Upload results from (1) to Cocoon
-  Future<void> sendResultsPath(String resultsPath) async {
-    final File resultFile = fs.file(resultsPath);
-    final Map<String, dynamic> resultsJson = json.decode(await resultFile.readAsString()) as Map<String, dynamic>;
-    await _sendUpdateTaskRequest(resultsJson);
-  }
-
-  /// Send [TaskResult] to Cocoon.
-  // TODO(chillers): Remove when sendResultsPath is used in prod. https://github.com/flutter/flutter/issues/72457
-  Future<void> sendTaskResult({
-    required String builderName,
-    required TaskResult result,
-    required String gitBranch,
+  ///
+  /// The `resultsPath` is not available for all tests. When it doesn't show up, we
```

Comment:
In what case do we not have resultsPath?

Reply:
Based on current infra configs, seems only devicelab tasks needing a physical device set `upload_metrics: true`. We have other devicelab hostonly tests that do not generate results.